magenta-lib/src/main/scala/magenta/tasks/tasks.scala

```diff
@@ -42,6 +42,7 @@ case class S3Upload(
     surrogateControlPatterns: List[PatternValue] = Nil,
     extensionToMimeType: Map[String, String] = Map.empty,
     publicReadAcl: Boolean = false,
+    lambdaArtifact: Boolean = false,
```

Comment:
The enclosing function is used by multiple deployment types; I wonder if we should be more generic in the naming here? For example `allowDeletionByLifecycleRule`. Maybe the tag could be generic too?

Reply:
Thanks for the suggestion, I think that's a good idea! https://github.com/guardian/riff-raff/pull/1207/commits/d002ee4f2feccc753b7c6c2c2b31f92f5e952767